src/index.js

```diff
@@ -3,55 +3,57 @@
 const waterfall = require('async/waterfall')
 const reject = require('async/reject')
 const each = require('async/each')
-const EventEmitter = require('events').EventEmitter
-const debug = require('debug')
+const series = require('async/series')
+const map = require('async/map')
+const once = require('once')
 
 const CONSTANTS = require('./constants')
-const WantManager = require('./components/want-manager')
-const Network = require('./components/network')
-const DecisionEngine = require('./components/decision-engine')
+const WantManager = require('./want-manager')
+const Network = require('./network')
+const DecisionEngine = require('./decision-engine')
+const Notifications = require('./notifications')
+const logger = require('./utils').logger
 
-const log = debug('bitswap')
-log.error = debug('bitswap:error')
-
-/**
- *
- */
 class Bitswap {
   /**
    * Create a new bitswap instance.
    *
    * @param {Libp2p} libp2p
    * @param {Blockstore} blockstore
-   * @param {PeerBook} peerBook
    * @returns {Bitswap}
    */
-  constructor (libp2p, blockstore, peerBook) {
-    this.libp2p = libp2p
+  constructor (libp2p, blockstore) {
+    this._libp2p = libp2p
+    this._log = logger(this.peerInfo.id)
+
     // the network delivers messages
-    this.network = new Network(libp2p, peerBook, this)
+    this.network = new Network(libp2p, this)
 
     // local database
     this.blockstore = blockstore
 
-    this.engine = new DecisionEngine(blockstore, this.network)
+    this.engine = new DecisionEngine(this.peerInfo.id, blockstore, this.network)
 
     // handle message sending
-    this.wm = new WantManager(this.network)
+    this.wm = new WantManager(this.peerInfo.id, this.network)
 
     this.blocksRecvd = 0
     this.dupBlocksRecvd = 0
     this.dupDataRecvd = 0
 
-    this.notifications = new EventEmitter()
-    this.notifications.setMaxListeners(CONSTANTS.maxListeners)
+    this.notifications = new Notifications(this.peerInfo.id)
+  }
+
+  get peerInfo () {
+    return this._libp2p.peerInfo
   }
 
   // handle messages received through the network
   _receiveMessage (peerId, incoming, callback) {
     this.engine.messageReceived(peerId, incoming, (err) => {
       if (err) {
-        log('failed to receive message', incoming)
+        // TODO: Q: why do we just log and not return here?
+        this._log('failed to receive message', incoming)
```

Comment:
The reason is that we still want to process the message as much as possible, even if there was a problem in the decision engine. (Technically that err is always undefined atm, but that could change later down the line)